src/components/DisplayWeb3.tsx

```diff
@@ -13,51 +18,110 @@ interface ClientVersion {
     result: string
 }
 
-export const DisplayWeb3: React.FC<DisplayWeb3Props> = ({ web3Api }) => {
-    const [account, setAccount] = useState<string>()
+const DEFAULT_RECIPIENT = '0x286E20d6e20d12D5Ba06F8eC2dAe91B7725f9188'
 
+export const DisplayWeb3: React.FC<DisplayWeb3Props> = ({ title = 'Web3 Data', web3Api }) => {
+    // User State
+    const [account, setAccount] = useState<string>()
     const [balance, setBalance] = useState<string>()
-
     const [networkName, setNetworkName] = useState<string>()
-
     const [clientVersion, setClientVersion] = useState<ClientVersion>()
+    // Send
+    const [amountToSend, setAmountToSend] = useState<string>('0')
+    const [recipientAddress, setRecipientAddress] = useState<string>(null)
+    // Async
+    const [error, setError] = useState<Error>(null)
+    const [loading, setLoading] = useState<boolean>(false)
 
     useEffect(() => {
         const setter = async (): Promise<void> => {
-            const accountsProm = web3Api.getAccounts()
-            const networkNameProm = web3Api.getNetworkName()
-            const clientVersionProm = web3Api.web3.currentProvider.send('web3_clientVersion', [])
-
-            const [account] = await accountsProm
-
-            const [balance, networkName, clientVersion] = await Promise.all([
-                web3Api.getBalance(account),
-                networkNameProm,
-                clientVersionProm,
-            ])
-
-            ReactDOM.unstable_batchedUpdates(() => {
-                setAccount(account)
-                setBalance(balance)
-                setNetworkName(networkName)
-                setClientVersion(clientVersion)
-            })
+            try {
+                setError(null)
+                setLoading(true)
+
+                const accountsProm = web3Api.getAccounts()
+                const networkNameProm = web3Api.getNetworkName()
+                const clientVersionProm = web3Api.web3.currentProvider.send('web3_clientVersion', [])
+
+                const [account] = await accountsProm
+
+                const [balance, networkName, clientVersion] = await Promise.all([
```

Comment:
How is the error being handled? Maybe is not super relevant for this component since I understand is a test, but I think we always need to handle or propagate the errors.

Reply:
For now it can be handled user-side with `getDerivedStateFromError` in user-made component. But you are right, a callback prop `onError` would be useful